pontoon/base/static/js/fluent_interface.js

```diff
@@ -316,7 +316,8 @@ var Pontoon = (function (my) {
             }
 
             if (value) {
-              value = '{ $num ->' + value + '\n  }';
+              var entity_ast = fluentParser.parseEntry(entity.original);
+              value = '{ ' + entity_ast.value.elements[0].expression.id.name + ' ->' + value + '\n  }';
```

Comment:
This only works if the select expression is the first thing. Already the examples on playground break this.
Doing a full-depth analysis might be overdoing it, but I'd at least search the top-level:

```
entity_ast.value.elements.filter(node => node.type === 'Placable' && node.expression.type === ''SelectExpression')[0].expression.id.name
````

should work, AFAICT.

Reply:
Correct. Are you saying we have examples exposed for localization that break with this? If not, I'd rather ship this and address the issue of "non-first" select expressions in a followup patch.

The reason is https://github.com/mozilla/pontoon/commit/874cb0198775661b3194e3cca1330ddce164d3a9, which I'd like to revert and then add fixes on top of it before making things hard to merge. We need to make changes you're proposing in multiple places, because plurals support in FTL editor is for now very limited - to strings that have plural variants in source string (and nothing else).

Reply:
@flodolo, anything on your radar where en-US uses select expressions as a part?

Re the relanding of the source editor, I can see that this code snippet is right in the middle of the patch ;-)

Reply:
If I understand the question correctly, I think we're good. I can only think of selectors used for plural forms. BTW, these are all the files we have around:
https://github.com/mozilla/send/blob/master/public/locales/en-US/send.ftl
https://github.com/mozilla-services/screenshots/blob/master/locales/en-US/server.ftl
https://github.com/mozilla/testpilot/blob/master/locales/en-US/app.ftl
https://github.com/mozilla/testpilot/blob/master/locales/en-US/experiments.ftl

Only the first two have plural forms.

Reply:
I've checked the first two, and they're of the form

    foo = { $num ->
        }

so we should get along with this. We'll need a follow-up bug, though.